src/core/qgstessellator.h

```diff
@@ -51,9 +56,114 @@ class CORE_EXPORT QgsTessellator
+
+    /**
+     * Sets the origin point of the map.
+     * \since QGIS 4.0
+     */
+    void setOrigin( const QgsVector3D &origin );
+
+    /**
+     * Returns the origin point of the map.
+     * \since QGIS 4.0
+     */
+    QgsVector3D origin() const { return mOrigin; }
+
+    /**
+     * Sets scaling and the bounds of the input geometry coordinates.
+     * \since QGIS 4.0
+     */
+    void setBounds( const QgsRectangle &bounds );
+
+    /**
+     * Sets whether Z values are ignored (TRUE) or not (FALSE).
+     * By default, extrusion is disabled.
+     * \since QGIS 4.0
+     */
+    void setInputZValueIgnored( bool ignore );
+
+    /**
+     * Returns whether Z values are ignored (TRUE) or not (FALSE).
+     * By default, extrusion is disabled.
+     * \since QGIS 4.0
+     */
+    bool isZValueIgnored() const { return mInputZValueIgnored; }
+
+    /**
+     * Sets which faces should be generated during extrusion.
+     * \since QGIS 4.0
+     */
+    void setExtrusionFaces( Qgis::ExtrusionFaces faces );
+
+    /**
+     * Returns which faces are generated during extrusion.
+     * \since QGIS 4.0
+     */
+    Qgis::ExtrusionFaces extrusionFaces() const { return mExtrusionFaces; }
+
+    /**
+     * Sets the rotation of generated faces.
+     * \since QGIS 4.0
+     */
+    void setTextureRotation( float rotation );
+
+    /**
+     * Returns the rotation of generated faces.
+     * \since QGIS 4.0
+     */
+    float textureRotation() const { return mTextureRotation; }
+
+    /**
+     * Sets whether texture UV coordinates should be added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
+     */
+    void setAddTextureUVs( bool addTextureUVs );
+
+    /**
+     * Returns whether texture UV coordinates are being added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
+     */
+    bool hasTextureUVs() const { return mAddTextureCoords; }
+
+    /**
+     * Sets whether normals should be added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
+     */
+    void setAddNormals( bool addNormals );
+
+    /**
+     * Returns whether normals are being added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
      */
-    QgsTessellator( const QgsRectangle &bounds, bool addNormals, bool invertNormals = false, bool addBackFaces = false, bool noZ = false,
-                    bool addTextureCoords = false, int facade = 3, float textureRotation = 0.0f );
+    bool hasNormals() const { return mAddNormals; }
+
+    /**
+     * Sets whether back faces should be added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
+     */
+    void setAddBackFaces( bool addBackFaces );
+
+    /**
+     * Returns whether back faces are being added to the output data (TRUE) or not (FALSE).
+     * \since QGIS 4.0
+     */
+    bool hasBackFaces() const { return mAddBackFaces; }
```

Comment:
it would be good to have consistent name with the setter - here we miss the "Add" bit, which would indeed read a bit weird here. How about setBackFacesEnabled() / hasBackFacesEnabled() ?